lnwallet/chanfunding/coin_select.go

```diff
@@ -267,14 +277,30 @@ func CoinSelectUpToAmount(feeRate chainfee.SatPerKWeight, maxAmt,
 	selectedCoins, changeAmt, err := CoinSelect(
 		feeRate, maxAmt, dustLimit, coins,
 	)
+	if err == nil {
```

Comment:
This whole method is very hard to grasp without staring at it for a long time. Since this selects coins, it makes me very nervous.

So here two suggestions to streamline the code in this area.

1. Use one `switch` statement:

```
switch {
case err == nil:

case err.(type) == *ErrInsufficientFunds:

default:

}
```

2. Instead of setting the err to `errReservedValueInvalidated`, do the actual call to `CoinSelectSubtractFees` directly.

Reply:
Covered here: https://github.com/lightningnetwork/lnd/pull/6903/commits/60ddaa719a0e36f104304ffa9adb901764c9e314#diff-40ac2ca08026898bb3f9c383bb3ddf7c232b5d2a46c31c8673a80c17c87cf037R277